library/ssl_srv.c

```diff
@@ -3039,8 +3039,8 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
 
         MBEDTLS_SSL_DEBUG_MSG( 2, ( "ECDHE curve: %s", (*curve)->name ) );
 
-        if( ( ret = mbedtls_ecp_group_load( &ssl->handshake->ecdh_ctx.grp,
-                                       (*curve)->grp_id ) ) != 0 )
+        if( ( ret = mbedtls_ecdh_setup( &ssl->handshake->ecdh_ctx,
+                                        (*curve)->grp_id ) ) != 0 )
```

Comment:
Why did we need to change this to `mbedtls_ecdh_setup()`? Couldn't we keep calling `mbedtls_ecp_group_load()`? If not, we've broken the API in the default configuration.

Reply:
Because the new context the `grp` member has different type and this call wouldn't compile in the non-legacy case. It would still work in the default configuration, but we want this to work in both. I don't think we have broken the API in the default configuration with this.

Reply:
OK, so we only changed the API such that if you want to work with the new non-legacy option, you need to change to call `mbedtls_ecdh_setup()`. This is fine.